scheduler/data-flow/src/main/kotlin/io/seldon/dataflow/kafka/Pipeline.kt

```diff
@@ -93,12 +137,33 @@ class Pipeline(
             kafkaDomainParams: KafkaDomainParams,
             kafkaConsumerGroupIdPrefix: String,
             namespace: String,
-        ): Pipeline {
+        ): Pair<Pipeline?, PipelineStatus.Error?> {
             val (topology, numSteps) = buildTopology(metadata, steps, kafkaDomainParams)
             val pipelineProperties = localiseKafkaProperties(kafkaProperties, metadata, numSteps, kafkaConsumerGroupIdPrefix, namespace)
-            val streamsApp = KafkaStreams(topology, pipelineProperties)
-            logger.info("Create pipeline stream for name:${metadata.name} id:${metadata.id} version:${metadata.version} stream with kstream app id:${pipelineProperties[StreamsConfig.APPLICATION_ID_CONFIG]}")
-            return Pipeline(metadata, topology, streamsApp, kafkaDomainParams, numSteps)
+            var streamsApp : KafkaStreams? = null
+            var pipelineError: PipelineStatus.Error? = null
+            try {
+                streamsApp = KafkaStreams(topology, pipelineProperties)
+            } catch (e: StreamsException) {
+                pipelineError = PipelineStatus.Error(null)
+                    .withException(e)
+                    .withMessage("failed to initialize kafka streams app")
+                return null to pipelineError
```

Comment:
question: lacking kotlin knowledge but assuming this returns `(null, pipelineError)`

Reply:
Indeed, that's the relatively strange but terse way of building a pair. It gets slightly less strange if you think of the pair as mapping one element **to** another one.